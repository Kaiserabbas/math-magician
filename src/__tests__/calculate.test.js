import calculate from '../logic/calculate';

describe('calculate function', () => {
  it('resets calculator data on AC button', () => {
    const result = calculate({ total: '10', next: '5', operation: '+' }, 'AC');
    expect(result).toMatchSnapshot();
  });

  it('handles number input correctly', () => {
    const result = calculate({ total: '10', next: '5', operation: '+' }, '1');
    expect(result).toMatchSnapshot();
  });

  it('handles decimal point input correctly', () => {
    const result = calculate({ total: '10', next: '5', operation: '+' }, '.');
    expect(result).toMatchSnapshot();
  });

  it('handles = operation correctly', () => {
    const result = calculate({ total: '10', next: '5', operation: '+' }, '=');
    expect(result).toMatchSnapshot();
  });

  it('handles +/- operation correctly', () => {
    const result = calculate({ total: '10', next: '5', operation: '+' }, '+/-');
    expect(result).toMatchSnapshot();
  });

  it('handles consecutive number inputs correctly', () => {
    const result = calculate({ total: '10', next: '5', operation: '+' }, '2');
    expect(result).toMatchSnapshot();
  });

  it('handles operation after number input correctly', () => {
    const result = calculate({ total: '10', next: '5', operation: '+' }, '-');
    expect(result).toMatchSnapshot();
  });

  it('handles consecutive operations correctly', () => {
    const result = calculate({ total: '10', next: '5', operation: '+' }, '/');
    expect(result).toMatchSnapshot();
  });

  it('handles operation after = correctly', () => {
    const result = calculate({ total: '10', next: '5', operation: '+' }, '=');
    expect(result).toMatchSnapshot();
  });

  it('handles operation after +/- correctly', () => {
    const result = calculate({ total: '10', next: '5', operation: '+' }, '+/-');
    expect(result).toMatchSnapshot();
  });
});
