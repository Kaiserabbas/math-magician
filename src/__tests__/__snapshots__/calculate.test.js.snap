// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculate function handles +/- operation correctly 1`] = `
Object {
  "next": "-5",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate function handles = operation correctly 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "15",
}
`;

exports[`calculate function handles consecutive number inputs correctly 1`] = `
Object {
  "next": "52",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate function handles consecutive operations correctly 1`] = `
Object {
  "next": null,
  "operation": "/",
  "total": "15",
}
`;

exports[`calculate function handles decimal point input correctly 1`] = `
Object {
  "next": "5.",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate function handles number input correctly 1`] = `
Object {
  "next": "51",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate function handles operation after +/- correctly 1`] = `
Object {
  "next": "-5",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate function handles operation after = correctly 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "15",
}
`;

exports[`calculate function handles operation after number input correctly 1`] = `
Object {
  "next": null,
  "operation": "-",
  "total": "15",
}
`;

exports[`calculate function resets calculator data on AC button 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": null,
}
`;
